evm_loader/program/src/entrypoint.rs

```diff
@@ -450,47 +458,56 @@ fn do_finalize<'a>(program_id: &Pubkey, accounts: &'a [AccountInfo<'a>]) -> Prog
         return Err(ProgramError::InvalidArgument);
     }
 
-    let mut backend = SolanaBackend::new(program_id, accounts, clock_info)?;
-    debug_print!("  backend initialized");
-
-    let caller_ether = get_ether_address(program_id, backend.get_account_by_index(1), caller_info, signer_info, None).ok_or(ProgramError::InvalidArgument)?;
-
-    let config = evm::Config::istanbul();
-    let mut executor = StackExecutor::new(&backend, usize::max_value(), &config);
-    debug_print!("  executor initialized");
-
-    debug_print!("Execute transact_create");
+    let account_storage = RefCell::new(ProgramAccountStorage::new(program_id, accounts, accounts.last().unwrap())?);
 
-    let code_data = {
-        let data = program_info.data.borrow();
-        let (_unused, rest) = data.split_at(AccountData::SIZE);
-        let (code_len, rest) = rest.split_at(8);
-        let code_len = code_len.try_into().ok().map(u64::from_le_bytes).unwrap();
-        let (code, _rest) = rest.split_at(code_len as usize);
-        code.to_vec()
-    };
-
-    let program_account = SolidityAccount::new(program_info)?;
-
-    let exit_reason = executor.transact_create2(
-            caller_ether.0,
-            U256::zero(),
-            code_data,
-            H256::default(), usize::max_value()
-        );
-    debug_print!("  create2 done");    
+    let caller_ether = get_ether_address(program_id, account_storage.borrow().get_account_by_index(1), caller_info, signer_info, None).ok_or(ProgramError::InvalidArgument)?;
+    
+    let (exit_reason, result, applies_logs) = {
```

Comment:
Why do you need to collect `exit_reason`, `result`, and `applies_logs` here?

Reply:
They are used later. I wanted to isolate SolanaBackend and StackExecutor in own scope to avoid account storage borrowing problems because they use immutable account storage and its used as mutable later.